.github/workflows/terraform.yml

```
@@ -133,12 +139,21 @@ jobs:
         # Required in order to check exitcode.
         # https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#exit-codes-and-error-action-preference
         shell: bash {0}
+        env:
+          TFVARS_FILE: ${{ inputs.var_file }}
         run: |
-          terraform plan -input=false -out="$TF_PLAN_FILE" -detailed-exitcode
+          OPTIONAL_ARGS=()
+          if [[ "$TFVARS_FILE" != "null" ]]; then
+            OPTIONAL_ARGS+=(-var-file="${TFVARS_FILE}")
+          fi
+
+          terraform plan -input=false -out=tfplan -detailed-exitcode "${OPTIONAL_ARGS[@]}"
```

Also, undo this change:

```suggestion
          terraform plan -input=false -out="$TF_PLAN_FILE" -detailed-exitcode "${OPTIONAL_ARGS[@]}"
```